Jint/Runtime/ExpressionIntepreter.cs

```diff
@@ -851,7 +852,17 @@ public JsValue EvaluateCallExpression(CallExpression callExpression)
 
             if (!func.IsObject())
             {
-                throw new JavaScriptException(_engine.TypeError, r == null ? "" : string.Format("Property '{0}' of object is not a function", (callee as Reference).GetReferencedName()));
+                var reference = (Reference) callee;
```

Comment:
This fails some tests with `System.InvalidCastException: Unable to cast object of type 'Jint.Native.JsValue' to type 'Jint.Runtime.References.Reference'.`

Reply:
This might have been hidden in the previous code because it was already throwing an exception. I assume the correct behavior is to throw a TypeError if it's not a reference.

Reply:
Yes, I assumed that this is correct because it would NRE otherwise. Will fix in a bit.

Reply:
Nope, there was no NRE, there is a check for `r == null` there.